coalib/parsing/LineParser.py

```diff
@@ -125,10 +127,19 @@ def __separate_by_first_occurrence(string,
         if return_second_part_nonempty and delim_pos == len(string):
             return "", string.strip(" \n")
 
-        return (
-            string[:delim_pos].strip(" \n"),
-            string[delim_pos + (len(used_delim) if strip_delim else 0):].strip(
-                " \n"))
+        first_part = string[:delim_pos]
+        second_part = string[delim_pos + (
+            len(used_delim) if strip_delim else 0):]
+
+        first_part_stripped = unescaped_rstrip(first_part)
+        second_part_stripped = unescaped_rstrip(second_part)
+
+        if position_is_escaped(second_part, len(second_part) - 1):
+            return (first_part.lstrip().strip("\n"),
+                    second_part.lstrip().strip("\n"))
```

Comment:
\+ to avoid duplication with `lstrip().strip("\n")`, you could reassign `first_part` and `second_part`

```python
if not position_is_escaped(...):
    first_part = unescaped_rstrip(first_part)
    second_part = unescaped_rstrip(second_part)

return (first_part.lstrip().strip("\n"), second_part.lstrip().strip("\n"))
```